mapserv.c

```diff
@@ -239,12 +238,33 @@ int main(int argc, char *argv[])
   signal( SIGTERM, msCleanupOnSignal );
 #endif
 
+  // make a copy of the original environment
+  int i = 0;
+  int j;
+  char **env = environ;
+
+  while (env[i] != NULL)
+      i++;
+
+  char **envCopy = malloc(sizeof(char*) * i + 1);
+
+  for (j = 0; j < i; j++) {
+      envCopy[j] = malloc(strlen(env[j]));
```

Comment:
but here you need it
```suggestion
      envCopy[i] = malloc(strlen(env[i])+1);
```